caiman/fiola/signal_analysis_online.py

```diff
@@ -0,0 +1,460 @@
+        Trace removed from subthreshold.
+    t_s : ndarray
+        Trace after template matching.
+    sub : ndarray
+        Trace of subtrehsold.
+    median : float
+        Median of the original trace
+    scale : float
+        Scale of the original trace based on 1 percentile.
+    thresh_factor : float
+        A value determine the threshold level.
+    median2 : float
+        Median of the trace after template matching. 
+    std: float
+        Estimated standard deviation
+    peak_to_std: float
+        Spike height devided by std
+    peak_level: float
+        95 percentile of the peak_to_std. 
+        It is updated to compensate for the photobleaching effect.
+    """
+    # Normalize and remove subthreshold
+    median = np.median(img)
+    scale = -np.percentile(img-median, 1)
+    if do_scale == True:
+        t0 = (img - median) / scale
+    else:
+        t0 = img.copy() - median
+    
+    #sub = signal_filter(t0, freq, fr, order=1, mode='low')
+    #filt_window=15
+    if type(filt_window) is list:
+        sub = non_symm_median_filter(t0, filt_window)                
+    else:
+        sub = median_filter(t0, size=filt_window)
+    t = t0 - sub
+
+    # First time thresholding
+    data = t.copy()
+    if do_plot:
+        plt.figure(); plt.plot(data);plt.show()
+    pks = data[signal.find_peaks(data, height=None)[0]]
+    thresh, _, _, _ = adaptive_thresh(pks, clip=100, pnorm=0.25, min_spikes=10)
+    idx = signal.find_peaks(data, height=thresh)[0]
+    
+    # Form a template
+    if template_window > 0:
+        window_length = template_window
+    else:
+        window_length = 2 
+    window = np.int64(np.arange(-window_length, window_length + 1, 1))
+    idx = idx[np.logical_and(idx > (-window[0]), idx < (len(data) - window[-1]))]
+    PTD = data[(idx[:, np.newaxis] + window)]
+    PTA = np.median(PTD, 0)
+    
+    if template_window > 0:
+        # Template matching, second time thresholding
+        t_s = np.convolve(data, np.flipud(PTA), 'same')
+        data = t_s.copy()
+    else:
+        logging.info('skip template matching')
+        data = t.copy()
+        t_s = t.copy()
+
+    median2 = np.median(data)
+    data = data - median2
+    if robust_std:
+        std = -np.percentile(data, 25) * 2 /1.35
+    else:
+        ff1 = -data * (data < 0)
+        Ns = np.sum(ff1 > 0)
+        std = np.sqrt(np.divide(np.sum(ff1**2), Ns)) 
+        
+    # Select best threshold based on estimated false positive rate
+    if adaptive_threshold:
+        pks2 = t_s[signal.find_peaks(t_s, height=None)[0]]
+        try:
+            thresh2, falsePosRate, detectionRate, low_spikes = adaptive_thresh(pks2, clip=0, pnorm=0.5, min_spikes=10)  # clip=0 means no clipping
+            thresh_factor = thresh2 / std
+            if thresh_factor < minimal_thresh:
+                logging.info(f'Adaptive threshold factor is lower than minimal theshold: {minimal_thresh}, choose thresh factor to be {minimal_thresh}')
```

Comment:
Should this be an error or a warning? If it's just info people will usually not notice.